src/py42/services/_connection.py

```diff
@@ -227,12 +226,14 @@ def _prepare_request(
         if json is not None:
             data = json_lib.dumps(json)
 
-        user_headers = _create_user_headers(headers, data)
-        self._headers.update(user_headers)
+        headers = _create_user_headers(headers, data)
```

Comment:
The user headers should be added last so that a method can manually override `Content-Type` (or any of the items in `self._headers`).